src/types/SisoTfTypes/SisoZpk.jl

```
@@ -6,19 +6,20 @@ struct SisoZpk{T,TR<:Number} <: SisoTf{T}
     z::Vector{TR}
     p::Vector{TR}
     k::T
-    function SisoZpk{T,TR}(z::Vector{TR}, p::Vector{TR}, k::T) where  {T<:Number, TR<:Number}
+    function SisoZpk{T,TR}(z::Vector{TR}, p::Vector{TR}, k::T) where {T<:Number, TR<:Number}
         if k == zero(T)
             p = TR[]
             z = TR[]
         end
         if TR <: Complex && T <: Real
-            @assert check_real(z) "zpk model should be real-valued, but zeros do not come in conjugate pairs."
-            @assert check_real(p) "zpk model should be real-valued, but poles do not come in conjugate pairs."
+            z, p = copy(z), copy(p)
+            @assert pairup_conjugates!(z) "zpk model should be real-valued, but zeros do not come in conjugate pairs."
```

Should we just let the error be thrown in `pairup_conjugates!` (similarly to in `common_timeevol`), or is this the way we want to go? This way we can let the user know if the problem is with the zeros or the poles. @mfalt

---

Was thinking about it when I did this, I originally wanted to move it to `pairup_conjugates!` which felt like nicer code, but then the error message would not be as informative so I thought this was better after all. But I'm open for suggestions.